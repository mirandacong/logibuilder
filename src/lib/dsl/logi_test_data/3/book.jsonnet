local defs = import '../defs.jsonnet';
local sheet1 = import './sheet1.jsonnet';
local sheet2 = import './sheet2.jsonnet';

local book = defs.Book('Test model', {
    subnodes: [sheet1, sheet2],
});

book
