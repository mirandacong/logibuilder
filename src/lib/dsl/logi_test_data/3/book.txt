@0,0,0,0@
```(
book Test model
    sheet Control

        table Revenue Breakup
            header period
                stub us mn
                2017 | 2018 | 2019 | 2020
                    Q1 | Q2 | Q3 | Q4 | FY

            row1 = source()
            row2 = source()
            row3 = source()

            row4 = {row1} + {row2}

            row5 = sum({row1}, {row2}, {row3})

            row6 = average({row1}, {row2}, {row3})
            row7 = input()
            row8 = 1
            row9 = {row1: Test model/sheet1/Revenue Breakup}
    sheet sheet1

        table Revenue Breakup
            header period
                stub us mn
                2017 | 2018 | 2019 | 2020
                    Q1 | Q2 | Q3 | Q4 | FY

            row1 = source()
)```
