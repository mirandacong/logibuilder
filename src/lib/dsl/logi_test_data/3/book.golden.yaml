---
annotations: {}
labels: []
name: Test model
nodetype: 1
subnodes:
- annotations: {}
  labels: []
  name: Control
  nodetype: 2
  subnodes:
  - annotations: {}
    header_stub: us mn
    labels: []
    name: Revenue Breakup
    nodetype: 4
    subnodes:
    - annotations: {}
      labels: []
      name: '2017'
      nodetype: 6
      subnodes:
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q1
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q2
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q3
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q4
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: FY
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
    - annotations: {}
      labels: []
      name: '2018'
      nodetype: 6
      subnodes:
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q1
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q2
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q3
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q4
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: FY
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
    - annotations: {}
      labels: []
      name: '2019'
      nodetype: 6
      subnodes:
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q1
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q2
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q3
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q4
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: FY
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
    - annotations: {}
      labels: []
      name: '2020'
      nodetype: 6
      subnodes:
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q1
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q2
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q3
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q4
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: FY
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
    - alias: ''
      annotations: {}
      expression: ''
      is_def_scalar: false
      labels: []
      modifier: ''
      name: row1
      nodetype: 7
      slice_exprs: []
      sources:
      - 1
      - 2
      - 3
      - 4
      - 5
      - 6
      - 7
      - 8
      - 9
      - 10
      - 11
      - 12
      - 13
      - 14
      - 15
      - 16
      - 17
      - 18
      - 19
      - 20
      subnodes: []
      type: 0
    - alias: ''
      annotations: {}
      expression: ''
      is_def_scalar: false
      labels: []
      modifier: ''
      name: row2
      nodetype: 7
      slice_exprs: []
      sources:
      - 1
      - 2
      - 3
      - 4
      - 5
      - 6
      - 7
      - 8
      - 9
      - 10
      - 11
      - 12
      - 13
      - 14
      - 15
      - 16
      - 17
      - 18
      - 19
      - 20
      subnodes: []
      type: 0
    - alias: ''
      annotations: {}
      expression: ''
      is_def_scalar: false
      labels: []
      modifier: ''
      name: row3
      nodetype: 7
      slice_exprs: []
      sources:
      - 1
      - 2
      - 3
      - 4
      - 5
      - 6
      - 7
      - 8
      - 9
      - 10
      - 11
      - 12
      - 13
      - 14
      - 15
      - 16
      - 17
      - 18
      - 19
      - 20
      subnodes: []
      type: 0
    - alias: ''
      annotations: {}
      expression: '{row1} + {row2}'
      is_def_scalar: false
      labels: []
      modifier: ''
      name: row4
      nodetype: 7
      slice_exprs: []
      sources: []
      subnodes: []
      type: 0
    - alias: ''
      annotations: {}
      expression: sum({row1}, {row2}, {row3})
      is_def_scalar: false
      labels: []
      modifier: ''
      name: row5
      nodetype: 7
      slice_exprs: []
      sources: []
      subnodes: []
      type: 0
    - alias: ''
      annotations: {}
      expression: average({row1}, {row2}, {row3})
      is_def_scalar: false
      labels: []
      modifier: ''
      name: row6
      nodetype: 7
      slice_exprs: []
      sources: []
      subnodes: []
      type: 0
    - alias: ''
      annotations: {}
      expression: ''
      is_def_scalar: false
      labels: []
      modifier: ''
      name: row7
      nodetype: 7
      slice_exprs: []
      sources:
      - 1
      - 2
      - 3
      - 4
      - 5
      - 6
      - 7
      - 8
      - 9
      - 10
      - 11
      - 12
      - 13
      - 14
      - 15
      - 16
      - 17
      - 18
      - 19
      - 20
      subnodes: []
      type: 0
    - alias: ''
      annotations: {}
      expression: '1'
      is_def_scalar: false
      labels: []
      modifier: ''
      name: row8
      nodetype: 7
      slice_exprs: []
      sources: []
      subnodes: []
      type: 0
    - alias: ''
      annotations: {}
      expression: '{sheet1!Revenue Breakup!row1}'
      is_def_scalar: false
      labels: []
      modifier: ''
      name: row9
      nodetype: 7
      slice_exprs: []
      sources: []
      subnodes: []
      type: 0
    - alias: ''
      annotations: {}
      expression: ''
      is_def_scalar: false
      labels: []
      modifier: --
      name: row10
      nodetype: 7
      slice_exprs: []
      sources:
      - -200
      - -112
      -
      -
      -
      -
      -
      -
      -
      -
      -
      -
      -
      -
      -
      -
      -
      -
      -
      -
      subnodes: []
      type: 0
- annotations: {}
  labels: []
  name: sheet1
  nodetype: 2
  subnodes:
  - annotations: {}
    header_stub: us mn
    labels: []
    name: Revenue Breakup
    nodetype: 4
    subnodes:
    - annotations: {}
      labels: []
      name: '2017'
      nodetype: 6
      subnodes:
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q1
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q2
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q3
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q4
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: FY
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
    - annotations: {}
      labels: []
      name: '2018'
      nodetype: 6
      subnodes:
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q1
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q2
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q3
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q4
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: FY
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
    - annotations: {}
      labels: []
      name: '2019'
      nodetype: 6
      subnodes:
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q1
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q2
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q3
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q4
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: FY
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
    - annotations: {}
      labels: []
      name: '2020'
      nodetype: 6
      subnodes:
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q1
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q2
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q3
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: Q4
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
      - annotations: {}
        expression: ''
        labels: []
        modifier: ''
        name: FY
        nodetype: 8
        slice_exprs: []
        sources: []
        subnodes: []
        type: 0
    - alias: ''
      annotations: {}
      expression: ''
      is_def_scalar: false
      labels: []
      modifier: ''
      name: row1
      nodetype: 7
      slice_exprs: []
      sources: []
      subnodes: []
      type: 0
